src/components/ProjectSettingsTabs/PluginsTab/getFormSchema.ts

```diff
@@ -0,0 +1,189 @@
+import { Field } from "@rjsf/core";
+import { FormDataProps, SpruceFormProps } from "components/SpruceForm";
+import { CardFieldTemplate } from "components/SpruceForm/FieldTemplates";
+import widgets from "components/SpruceForm/Widgets";
+import { hiddenIf, placeholderIf, radioBoxOptions } from "../utils";
+import { FormState } from "./types";
+
+export const getFormSchema = (
+  useRepoSettings: boolean,
+  repoData?: FormState,
+  formData?: FormDataProps
+): {
+  fields: Record<string, Field>;
+  schema: SpruceFormProps["schema"];
+  uiSchema: SpruceFormProps["uiSchema"];
+} => ({
+  fields: {},
+  schema: {
+    type: "object" as "object",
+    properties: {
+      performanceSettings: {
+        type: "object" as "object",
+        title: "Performance Plugins",
+        properties: {
+          perfEnabled: {
+            type: ["boolean", "null"],
+            title: "",
+            oneOf: radioBoxOptions(
+              ["Enabled", "Disabled"],
+              repoData?.performanceSettings?.perfEnabled
+            ),
+          },
+        },
+      },
+      buildBaronSettings: {
+        type: "object" as "object",
+        title: "Build Baron and Task Annotations",
+        properties: {
+          taskAnnotationSettings: {
+            title: "",
+            type: "object" as "object",
+            properties: {
+              jiraCustomFields: {
+                type: "array" as "array",
+                title: "Custom Jira Fields",
+                items: {
+                  type: "object" as "object",
+                  properties: {
+                    field: {
+                      type: "string" as "string",
+                      title: "Field",
+                    },
+                    displayText: {
+                      type: "string" as "string",
+                      title: "Display Text",
+                    },
+                  },
+                },
+              },
+            },
+          },
+          useBuildBaron: {
+            type: "boolean" as "boolean",
+            oneOf: radioBoxOptions(
+              [
+                "Build Baron Ticket Search and Create",
+                "Custom Ticket Creation",
+              ],
+              undefined
+            ),
+          },
+          ticketSearchProjects: {
+            type: "array" as "array",
+            title: "Ticket Search Projects",
```

Comment:
In the old UI it's required that Ticket Search Projects and Ticket Create Project both be populated if one is populated. Should that be maintained?

Reply:
I was undecided about that. I don't see why one should depend on the other since one is used for the file ticket webhook, while the other is used to populate the "related tickets" list, they don't seem to be dependent on each other. However, since nobody has asked to be able to do one without the other, we have no reason to break that pattern now.

Reply:
If we do indeed decide to make it not required, we would need a one-line change to the backend function that checks is the build baron is configured.

Reply:
I'm concerned that making it not required might also confuse other places that assume one exists without the other (the task page for example?) so that may be worth making as a different ticket but I defer to your decision here.

Reply:
right, that's why we would need to change the function that checks if the build baron is configured, which is what the other pages use. 
We have no reason to change it though, so I'll re-implement making them both required.